sourcegraph/sourcegraph
# batches: allow non-admins to view rollout window configuration

Closes #52901

Earlier, when we surfaced the rollout window config for non-site admins, we didn't test this feature with a non-site-admin account; this led to non-site admins seeing the error below.

![image](https://github.com/sourcegraph/sourcegraph/assets/25608335/08018eea-245e-4b5c-ac5a-27fd7a3e0812)

This is expected because the graphql query shown below, which we use to fetch the rollout window configuration, only allows admins to fetch the config.

```graphql
query {
  site {
    configuration { ... }
  }
}
```

This PR adds an argument to the `configuration` field, which will return a list of whitelisted configs that can be displayed to the user.

## Test plan


*  Configure rollout windows on your Sourcegraph instance `batchChanges.rolloutWindows` as a site-admin.
* Log in as a non-site-admin
* Access the Batch Changes User settings `/users/bool/settings/batch-changes`
* You should be able to view the rollout window configured.
<img width="1154" alt="CleanShot 2023-06-27 at 14 45 10@2x" src="https://github.com/sourcegraph/sourcegraph/assets/25608335/2de0387a-a3ef-43b9-a937-5fad49a98cd0">
